src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/ChainPal.cs

```diff
@@ -2,21 +2,34 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using Internal.Cryptography.Pal.Native;
+using Microsoft.Win32.SafeHandles;
 using System;
 using System.Runtime.InteropServices;
 using System.Security.Cryptography;
 using System.Security.Cryptography.X509Certificates;
 
-using Internal.Cryptography.Pal.Native;
-
-using Microsoft.Win32.SafeHandles;
-
 namespace Internal.Cryptography.Pal
 {
     internal sealed partial class ChainPal : IDisposable, IChainPal
     {
+        private SafeX509ChainHandle _chain;
+
+        public static IChainPal FromHandle(IntPtr chainContext)
+        {
+            if (chainContext == IntPtr.Zero)
+                throw new ArgumentNullException(nameof(chainContext));
+
+            SafeX509ChainHandle certChainHandle = Interop.crypt32.CertDuplicateCertificateChain(chainContext);
+            if (certChainHandle == null || certChainHandle == SafeX509ChainHandle.InvalidHandle)
```

Comment:
Why compare it against a static InvalidHandle property, vs call the instance IsInvalid property?